src/reducer/database/osm/reconstructor.ts

```diff
@@ -0,0 +1,260 @@
+        success: false,
+        message: err.message,
+      };
+    }
+  }
+
+  /**
+   * Get filtered objects of the table specified by type of Base.
+   *
+   * @param filter A filter set
+   * @param Cls A optional model. When not provided Base is chosen
+   * @param filterStatement A optional filter statement. When not provided .all()
+   *                       is chosen.
+   * @returns {object[]} A list of filtered objects of the initialized Base class
+   * @description Get all objects converted to type Cls or Base in a list
+   */
+  filter(filter: any, Cls?: any, filterStatement?: string) {
+    try {
+      if (!filterStatement) {
+        filterStatement = this.Base.statements.all;
+      }
+
+      let response = SOAssembler.database.exec(filterStatement);
+      for (let entry in response) {
+        if ({}.hasOwnProperty.call(response, entry)) {
+          for (let f in filter) {
+            if ({}.hasOwnProperty.call(filter, f)) {
+              /**
+               * Proxy to neutralize Generic Object Injection Sink.
+               *
+               * @todo Validate the proxy workaround for the injection
+               *       vulnerability via a penetration test.
+               * @see {@link https://bit.ly/2KdpgAh |the-dangers-of-square-bracket-notation}
+               * @see {@link https://bit.ly/3cpPVG6 |creating-defensive-objects-with-es6-proxies}
+               */
+              var eProxy = new Proxy(
+                { entry },
+                {
+                  get: () => {
+                    return entry;
+                  },
+                }
+              );
+
+              /**
+               * Proxy to neutralize Generic Object Injection Sink.
+               *
+               * @todo Validate the proxy workaround for the injection
+               *       vulnerability via a penetration test.
+               * @see {@link https://bit.ly/2KdpgAh |the-dangers-of-square-bracket-notation}
+               * @see {@link https://bit.ly/3cpPVG6 |creating-defensive-objects-with-es6-proxies}
+               */
+              var fProxy = new Proxy(
+                { f },
+                {
+                  get: () => {
+                    return f;
+                  },
+                }
+              );
+
+              if (response[eProxy.entry]) {
+                if (filter[fProxy.f] !== response[eProxy.entry][fProxy.f]) {
+                  delete response[eProxy.entry];
+                }
+              }
+            }
+          }
+        }
+      }
+
+      var filtered = response.filter((el: any) => {
+        return el !== null;
+      });
+
+      let objects = filtered.map((entry: any) => {
+        let o;
+
+        if (Cls) {
+          o = new Cls(entry);
+        } else {
+          o = new this.Base(entry);
+        }
+
+        return o;
+      });
+
+      return objects;
+    } catch (err) {
+      //#LEGACY
+      //#ERROR
+      console.error("OSM reconstructor .filer()", err);
```

Comment:
Should this be removed before merging to master, because it is legacy?
```suggestion
```

Reply:
```suggestion
      //#LEGACY
      //#TODO Implement proper error handling
      //#ERROR
      console.error("OSM reconstructor .filer()", err);
```